Marlin/src/lcd/extui/anycubic_vyper/dgus_tft.cpp

```diff
@@ -1001,34 +994,25 @@ namespace Anycubic {
   }
 
   void DgusTFT::ProcessPanelRequest() {
-    unsigned char * p_u8;
-    unsigned int control_index = 0;
-    unsigned int control_value;
-    unsigned int temp;
+    uint16_t control_index = 0;
+    uint32_t control_value;
+    uint16_t temp;
     char str_buf[20];
 
     if (data_received) {
       data_received = false;
 
       if (0x83 == data_buf[0]) {
-        p_u8 = (unsigned char *)(&control_index);// get control address
-        p_u8[0] = data_buf[2];
-        p_u8[1] = data_buf[1];
-
-        if ((control_index & 0xF000) == KEY_ADDRESS) {// is KEY
+        control_index = uint16_t(data_buf[1] << 8) | uint16_t(data_buf[2]);
+        if (control_index == KEY_ADDRESS) { // is KEY
```

Comment:
@Bob-the-Kuhn — This is clearly my error. It should be `(control_index & 0xF000) == KEY_ADDRESS`. Hopefully, my _only_ error!